### PR TITLE
Removed direct jackson dependencies from azure cosmos and spark

### PR DESCRIPTION
**Pull request by kushagraThapar**
* This PR removes direct jackson dependencies from azure-cosmos module. azure-cosmos depends on azure-core for jackson dependencies. However, to solve a CVE, we added direct jackson dependencies to azure-cosmos since azure-core was already released earlier. 
* Now that azure-core has updated jackson dependencies, we can remove the overridden ones.
* Reference PR - https://github.com/Azure/azure-sdk-for-java/pull/41329

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-spark_3-5_2-12/pom.xml [hunk 1]
```diff
@@ -177,22 +177,12 @@
...
-    <dependency>
-      <groupId>com.fasterxml.jackson.module</groupId>
-      <artifactId>jackson-module-afterburner</artifactId>
-      <version>2.15.2</version> <!-- {x-version-update;cosmos_com.fasterxml.jackson.module:jackson-module-afterburner;external_dependency} -->
+      <version>2.17.2</version> <!-- {x-version-update;com.fasterxml.jackson.core:jackson-databind;external_dependency} -->
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.module</groupId>
       <artifactId>jackson-module-scala_2.12</artifactId>
-      <version>2.15.2</version> <!-- {x-version-update;cosmos_com.fasterxml.jackson.module:jackson-module-scala_2.12;external_dependency} -->
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos-spark_3-5_2-12/pom.xml:195]: Shouldn't this be removed now from external_dependencies.txt as they aren't being used anymore?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml [hunk 2]
```diff
@@ -240,13 +240,9 @@
                 <include>org.apache.maven.plugins:maven-antrun-plugin:[3.1.0]</include> <!-- {x-include-update;org.apache.maven.plugins:maven-antrun-plugin;external_dependency} -->
                 <include>net.alchim31.maven:scala-maven-plugin:[4.8.1]</include> <!-- {x-include-update;cosmos_net.alchim31.maven:scala-maven-plugin;external_dependency} -->
                 <include>org.scalastyle:scalastyle-maven-plugin:[1.0.0]</include> <!-- {x-include-update;cosmos_org.scalastyle:scalastyle-maven-plugin;external_dependency} -->
-                <include>com.fasterxml.jackson.core:jackson-databind:[2.15.2]</include> <!-- {x-include-update;cosmos_com.fasterxml.jackson.core:jackson-databind;external_dependency} -->
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml:243]: Same with all these shouldn't this be removed now from external_dependencies.txt ?

**Review by tvaron3 — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos-spark_3-5_2-12/pom.xml (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos-spark_3-5_2-12/pom.xml:195]: Great question, they are all used in spark data resolver sample pom.xml
I think we should be able to remove them, but not sure.

File: sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml:243]: Great question, they are all used in spark data resolver sample pom.xml
I think we should be able to remove them, but not sure.

**Pull request merged**